homeassistant/components/fritzbox/__init__.py

```diff
@@ -84,8 +106,35 @@ async def async_setup_entry(hass, entry):
     except LoginError as err:
         raise ConfigEntryAuthFailed from err
 
-    hass.data.setdefault(DOMAIN, {CONF_CONNECTIONS: {}, CONF_DEVICES: set()})
-    hass.data[DOMAIN][CONF_CONNECTIONS][entry.entry_id] = fritz
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = {
+        CONF_CONNECTIONS: fritz,
+    }
+
+    async def async_update_coordinator():
+        """Fetch all device data."""
+        data = {}
+        for device in await hass.async_add_executor_job(fritz.get_devices):
+            device: FritzhomeDevice
+            try:
+                await hass.async_add_executor_job(device.update)
+            except requests.exceptions.HTTPError as ex:
+                await hass.async_add_executor_job(fritz.login)
```

Comment:
If we except the login to be successful, try the update again before raising `UpdateFailed`, otherwise raise `ConfigEntryAuthFailed` to stop the coordinator and start the reauth flow

Reply:
ok, now I get it ... thanks for being patient.
Because `fritz.get_devices()` does already I/O with the API, the try-except block is now already around this one.